packages/lodestar/src/network/attestationService.ts

```diff
@@ -0,0 +1,400 @@
+import {computeSubnetForCommitteesAtSlot} from "@chainsafe/lodestar-beacon-state-transition";
+import {IBeaconConfig, ForkName} from "@chainsafe/lodestar-config";
+import {ATTESTATION_SUBNET_COUNT, phase0, Slot, ValidatorIndex} from "@chainsafe/lodestar-types";
+import {assert, ILogger, randBetween} from "@chainsafe/lodestar-utils";
+import {ChainEvent, IBeaconChain} from "../chain";
+import {Eth2Gossipsub, GossipType} from "./gossip";
+import {MetadataController} from "./metadata";
+import {SubnetMap} from "./peers/utils";
+
+/// The time (in slots) before a last seen validator is considered absent and we unsubscribe from the random
+/// gossip topics that we subscribed to due to the validator connection.
+const LAST_SEEN_VALIDATOR_TIMEOUT = 150;
+
+export interface IAttestationService {
+  validatorSubscriptions(subscriptions: phase0.BeaconCommitteeSubscription[]): void;
+  shouldProcessAttestation(subnet: number, slot: phase0.Slot): boolean;
+  getActiveSubnets(): number[];
+}
+
+export interface IAttestationServiceModules {
+  config: IBeaconConfig;
+  chain: IBeaconChain;
+  logger: ILogger;
+  gossip: Eth2Gossipsub;
+  metadata: MetadataController;
+}
+
+/**
+ * Manage random (long lived) subnets and committee (short lived) subnets.
+ */
+export class AttestationService implements IAttestationService {
+  private readonly config: IBeaconConfig;
+  private readonly chain: IBeaconChain;
+  private readonly gossip: Eth2Gossipsub;
+  private readonly metadata: MetadataController;
+  private readonly logger: ILogger;
+
+  // Map of random subnets and the slot until they are needed of current fork
+  private randomSubnets: SubnetMap;
+  // Map of random subnets and the slot until they are needed of next fork
+  // TODO: handle multiple hard forks?
+  private nextForkRandomSubnets: SubnetMap | undefined;
+  // Map of committee subnets and the slot until they are needed
+  private committeeSubnets: SubnetMap;
+  // subset of committeeSubnets with exact slot for aggregator
+  private subscribedCommitteeSubnets: SubnetMap;
```

Comment:
Might be helpful for these comments to be `/** */` comments so they show up as help text in IDE.